src/typeahead/typeahead-window.ts

```diff
@@ -39,6 +41,12 @@ export class NgbTypeaheadWindow implements OnInit {
   activeIdx = 0;
 
   /**
+   *  An optional id for the typeahead. The id should be unique.
+   *  If not provided, it will be auto-generated.
+   */
+  @HostBinding('attr.id') @Input() id: string = undefined;
```

Comment:
The `id` is actually generated in the typeahead.ts file and set when the popup is opened (see `_openPopup` method).
It's actually kinda hackish but I don't know how else I can set the `id` of the popup. If I generate it in the typeahead-window.ts file, a new `id` will be generated each time the popup is opened (since a new instance is created each time)

I applied your other comments.